server/rest/src/main/java/org/infinispan/rest/resources/ClusterResource.java

```diff
@@ -92,4 +99,12 @@ private CompletionStage<RestResponse> getAllRestoreNames(RestRequest request) {
    private CompletionStage<RestResponse> restore(RestRequest request) {
       return BackupManagerResource.handleRestoreRequest(request, backupManager, (name, path, json) -> backupManager.restore(name, path));
    }
+
+   private CompletionStage<RestResponse> distribution(RestRequest request) {
+      RestServer server = invocationHelper.getProtocolServer();
+      return CompletableFuture.supplyAsync(server::clusterDistribution, invocationHelper.getExecutor())
```

Comment:
nitpick: `CompletableFuture.supplyAsync` not required.

Reply:
Was unsure how to act, should just call `this.clusterDistribution()`?

Reply:
yes. the `clusterDistribution()` method does not block since only submits works to the cluster executor.